px-data-grid-toggle-details-column.html

```diff
@@ -0,0 +1,110 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../vaadin-grid/vaadin-grid-column.html">
+<link rel="import" href="../vaadin-checkbox/vaadin-checkbox.html">
+
+<dom-module id="px-data-grid-toggle-details-column">
+  <template>
+    <template class="header">
+      <!-- empty header -->
+    </template>
+
+    <template id="defaultBodyTemplate">
+      <style>
+        .toggle-details {
+          margin: 0;
+          padding: 0;
+          width: auto;
+          background: none;
+          border: none;
+        }
+
+        .toggle-details:focus {
+          outline: none;
+        }
+
+        .toggle-details > px-icon {
+          padding: 0;
+          margin: 0;
+        }
+      </style>
+      <button class="toggle-details" on-tap="_onToggleTap">
+        <px-icon icon="[[_getChevronIcon(detailsOpened)]]"></px-icon>
+      </button>
+    </template>
+
+  </template>
+
+  <script>
+  {
+    /**
+     * `<px-data-grid-toggle-details-column>` is a helper element for the `<px-data-grid>`
+     * to be used to toggle row expansion (row details) column behavior.
+     * @memberof Predix
+     * @mixes Vaadin.GridColumnElement
+     */
+    class DataGridToggleDetailsColumnElement extends Vaadin.GridColumnElement {
+
+      static get is() {
+        return 'px-data-grid-toggle-details-column';
+      }
+
+      static get properties() {
+        return {
+          /**
```

Comment:
Add isDataColumn boolean property with value false, this can used to separate data columns from boiler plate columns. See selection px-data-grid-selection-column for example. This will prevent things some issues like this column showing up in action menu.